pkg/controller/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -427,6 +427,63 @@ func TestSyncMurStatusWithUserAccountStatuses(t *testing.T) {
 	logf.SetLogger(logf.ZapLogger(true))
 	s := apiScheme(t)
 
+	t.Run("to be provisioned notification created", func(t *testing.T) {
+		// given
+		// setup MUR that wil contain UserAccountStatusEmbedded fields for UserAccounts from "member2-cluster" but will miss from test.MemberClusterName
+		// then the reconcile should sync status and create a notification CR since the status is set to provisioned
+		mur := murtest.NewMasterUserRecord("john",
+			murtest.Finalizer("finalizer.toolchain.dev.openshift.com"),
+			murtest.StatusCondition(toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "")),
+			murtest.AdditionalAccounts("member2-cluster"))
+
+		userAccount := uatest.NewUserAccountFromMur(mur,
+			uatest.StatusCondition(toBeProvisioned()), uatest.ResourceVersion("123abc"))
+		userAccount2 := uatest.NewUserAccountFromMur(mur,
+			uatest.StatusCondition(toBeProvisioned()), uatest.ResourceVersion("123abc"))
+
+		mur.Status.UserAccounts = []toolchainv1alpha1.UserAccountStatusEmbedded{
+			{
+				SyncIndex: "111aaa",
+				Cluster: toolchainv1alpha1.Cluster{
+					Name: "member2-cluster",
+				},
+				UserAccountStatus: userAccount2.Status,
+			},
+		}
+
+		memberClient := test.NewFakeClient(t, userAccount, consoleRoute())
+		memberClient2 := test.NewFakeClient(t, userAccount2, consoleRoute())
+
+		hostClient := test.NewFakeClient(t, mur)
+		cntrl := newController(hostClient, s, newGetMemberCluster(true, v1.ConditionTrue),
+			clusterClient(test.MemberClusterName, memberClient), clusterClient("member2-cluster", memberClient2))
+
+		// when
+		_, err := cntrl.Reconcile(newMurRequest(mur))
+
+		// then
+		require.NoError(t, err)
+
+		uatest.AssertThatUserAccount(t, "john", memberClient).
+			Exists().
+			MatchMasterUserRecord(mur, mur.Spec.UserAccounts[0].Spec).
+			HasConditions(userAccount.Status.Conditions...)
+
+		murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+			HasStatusUserAccounts(test.MemberClusterName, "member2-cluster").
+			HasConditions(toBeProvisioned(), toBeProvisionedNotificationCreated()).
+			AllUserAccountsHaveStatusSyncIndex("123abc").
+			AllUserAccountsHaveCondition(userAccount.Status.Conditions[0])
+
+		// Get the notification resource
+		notification := toolchainv1alpha1.Notification{}
+		hostClient.Get(context.TODO(), namespacedName(mur.Namespace, userAccount.Name+"-provisioned"), &notification)
+		assert.Equal(t, notification.Spec.UserID, userAccount.Spec.UserID)
+		assert.Equal(t, notification.Spec.Template, "userprovisioned")
+		assert.Equal(t, notification.Name, userAccount.Name+"-provisioned")
+		assert.Equal(t, notification.Namespace, mur.Namespace)
```

Comment:
Can we just add these status and resource assertions to the existing test instead of copy-pasting it?

Reply:
fixed as part of: 43b2953aa3f7922169954600dd7d4cabfbe0a2ac